packages/flutter/lib/src/painting/shader_warm_up.dart

```diff
@@ -0,0 +1,144 @@
+import 'dart:developer';
+import 'dart:ui';
+
+/// Interface for drawing an image to warm up Skia shader compilations.
+///
+/// When Skia first sees a certain type of draw operations on GPU, it needs to
+/// compile the corresponding shader. The compilation can be slow (20ms-200ms).
+/// Having that time as a startup latency is much better than having a jank in
+/// the middle of an animation.
```

Comment:
> Having that time as a startup latency is much better than having a jank in the middle of an animation.

I think we should say it's "often" better, and make sure we provide a way for people to opt-out.